tests/test_amclient.py

```diff
@@ -922,6 +927,30 @@ def test_get_jobs(self):
         assert len(response) == 1
         assert response[0]["name"] == "Verify metadata directory checksums"
 
+    @mock.patch("utils.requests.request")
+    def test_get_status(self, mock_request):
+        mock_request.return_value.json.return_value = {"status": "PROCESSING"}
+        response = amclient.AMClient(
+            am_api_key=AM_API_KEY, am_user_name=AM_USER_NAME, am_url=AM_URL
+        ).get_unit_status("ca480d94-892c-4d99-bbb1-290698406571")
+        assert response.get("status") == "PROCESSING"
+
+        mock_request.return_value.json.return_value = {
+            "status": "COMPLETE",
+            "sip_uuid": "1234",
+        }
+        response = amclient.AMClient(
+            am_api_key=AM_API_KEY, am_user_name=AM_USER_NAME, am_url=AM_URL
+        ).get_unit_status("ca480d94-892c-4d99-bbb1-290698406571")
+        assert response.get("status") == "COMPLETE"
+
+        mock_request.return_value.ok = False
+        mock_request.return_value.json.return_value = {}
+        response = amclient.AMClient(
+            am_api_key=AM_API_KEY, am_user_name=AM_USER_NAME, am_url=AM_URL
+        ).get_unit_status("ca480d94-892c-4d99-bbb1-290698406571")
+        assert len(response) == 0
+
```

Comment:
Perhaps what's missing here are some call assertions (`assert_called`, `assert_called_with`, etc...).
E.g., when status=COMPLETE and sip_uuid=1234, we should expect the mock to have been called twice. Off the top of my head, I think that this is how assert_has_calls could help you:

```python
mock_request.assert_has_calls([
    mock.call('GET', url='/api/transfer/status/', ...),
    mock.call('GET', url='/api/ingest/status/', ...),
])

Reply:
OK, I've added some `assert_called_with` and `assert_called_once_with` tests which are definitely more useful than what I had previously asserted :)

Reply:
Awesome, I like it!